tools/clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -7473,6 +7473,14 @@ def err_hlsl_attribute_mismatch: Error<
   "invalid shader stage attribute combination">;
 def err_hlsl_attribute_valid_on_function_only: Error<
   "attribute is valid only on functions">;
+def err_hlsl_attribute_valid_on_entry_function_only: Error<
+  "%0 attribute only valid on entry point functions">;
+def err_hlsl_attr_for_wrong_shader: Error<
+  "attribute %0 only valid for %1.">;
```

Comment:
Maybe:

```
def err_hlsl_attribute_unsupported_stage : Error<
  "attribute %0 requires shader stage %1">;
```
Note no period at the end of the diagnostic message which is consistent with clang diagnostic styles.